### PR TITLE
plat/xen/arm: Set pg_count parameter for mrd regions

### PR DESCRIPTION
**Pull request by oleksiimoisieiev**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): `arm64`
 - Platform(s): `xen`
 - Application(s): N/A

### Description of changes

Set pg_count in _init_mem call to fix assertion failure on start:

<memory.c @   82> Assertion failure: (mrd)->pg_count * 0x1000UL == (mrd)->len

This assertion checks for mrd->pg_count parameter that was introduced in the following commit:
ad52a90f (uk/plat/memory: Introduce `pg_off` and `pg_count` memregion fields, 2023-10-28)

### CODE REVIEW

**Review by mogasergiu — CHANGES_REQUESTED** [review]
s/mfd/mrd

And what about the `pg_off` parameter?

**Comment by oleksiimoisieiev** [comment]
> s/mfd/mrd
> 
> And what about the `pg_off` parameter?

Yeah. thanks. pg_off parameter is 0 in my case. And it will be 0 by default because Xen scrubbing guest pages on start. But still I've added pg_off = 0 to improve code readability.
Thank you for the review

**Comment by michpappas** [comment]
Looks good to me, I will wait for @mogasergiu's RB before I merge.

**Review by mogasergiu — CHANGES_REQUESTED** [review]
Hey.. thanks for the fast response!

> And it will be 0 by default because Xen scrubbing guest pages on start

Correct, however should someone add another similar `mrd` initialization between those two with a `pg_off != 0`, the story might end up being different, with that person having to debug why the dtb region reuses a non-zero pg_off from the previous newly added mrd.

By the way, nitpick, can you use the `PAGE_COUNT` macro instead for `fdt_size` instead of the `PAGE_SHIFT` shift?

**Comment by razvand** [comment]
@oleksiimoisieiev , could you also use "Set" instead of "set" for your commit message, to pass the format check error?

**Comment by oleksiimoisieiev** [comment]
> Hey.. thanks for the fast response!
> 
> > And it will be 0 by default because Xen scrubbing guest pages on start
> 
> Correct, however should someone add another similar `mrd` initialization between those two with a `pg_off != 0`, the story might end up being different, with that person having to debug why the dtb region reuses a non-zero pg_off from the previous newly added mrd.
> 

Good point.


> By the way, nitpick, can you use the `PAGE_COUNT` macro instead for `fdt_size` instead of the `PAGE_SHIFT` shift?

It's great observation. Thank you for finding this. PAGE_COUNT is safer then shifting because it does DIV_ROUND_UP.

**Comment by oleksiimoisieiev** [comment]
> @oleksiimoisieiev , could you also use "Set" instead of "set" for your commit message, to pass the format check error?

Fixed. Sry... working on 5 repo right now and each has it's commit format...

**Review by mogasergiu — APPROVED** [review]
Thanks a lot for the quick update! :muscle: 

Reviewed-by: Sergiu Moga <sergiu@unikraft.io>

**Review by andreistan26 — APPROVED** [review]
Looks good, thanks!

Reviewed-by: Andrei Stan [andreistan2003@gmail.com](mailto:andreistan2003@gmail.com)

**Comment by michpappas** [comment]
Approved-by: Michalis Pappas <michalis@unikraft.io>

**Pull request merged**
